src/CattyTests/DataModel/VariablesContainerTest.swift

```diff
@@ -236,4 +236,41 @@ final class VariablesContainerTest: XCTestCase {
         XCTAssertTrue(lists[0].isList)
         XCTAssertTrue(lists[1].isList)
     }
+
+    func testIsEqualToVariablesContainer() {
```

Comment:
Please add a few other test methods to verify all possible code execution paths of this complex method (e.g. unequal number of variables/lists, variable with same name but different object, variable with same name but different values, ...).